tests/core/test_rules_engine.py

```
@@ -121,3 +232,95 @@ def check1(): return True, ""
+    bad_file.write_text("invalid python syntax {{{")  # Syntax error
+
+    config = CustomRulesConfig(paths=[str(bad_file)])
+    discover_rules(config)
+
+    captured = capsys.readouterr()
+    assert "Warning: Failed to load rule file" in captured.out
+    assert str(bad_file) in captured.out
+
+
+@patch("devrules.adapters.prompters.factory.get_default_prompter")
+def test_prompt_for_rule_arguments_rule_not_found(mock_get_prompter):
```

Ruff (0.14.13)

Unused function argument: `mock_get_prompter`

(ARG001)